### PR TITLE
Move the platform-dependent stuff in renderingengine.cpp to irrlicht

### PR DESCRIPTION
**Pull request by Desour**
Depends on https://github.com/minetest/irrlicht/pull/181. Edit: Was merged. PR is now waiting for an irrlicht revision bump.

## To do

This PR is a Ready for Review.

## How to test

* Start minetest.
* Look at the window icon.
* Look at the gui size (i.e. in the mainmenu).

I can only test with SDL or X11. The windows needs to be tested by someone else.

### CODE REVIEW

**Comment by sfan5** [comment]
We can stop trying to link to X11 then (`src/CMakeLists.txt`).

**Comment by SmallJoker** [comment]
@Desour Did you mean to mark this PR as "Ready for review", or is something missing to justify the "Draft" state?

**Comment by Desour** [comment]
Nothing is missing. But it's waiting on dependency. I was also hoping that github would not run the actions if it's still a draft, but that didn't work.
Marked now as ready, to avoid misunderstandings.

**Review by SmallJoker — APPROVED** [review]
Works. Tested on Linux/X11.

**Comment by Desour** [comment]
To my knowledge, there was nobody yet to test this on Windows.
Anyone with Windows can do this. Just fetch this PR's branch and the irrlicht PR's branch, recompile, and launch minetest. Then check if the window icon is set correctly and that the gui is scaled normally.

**Comment by sfan5** [comment]
Note that we can't merge this before the Irrlicht change is merged *and* we update the revision or switch to submodules.
(this is also why the builds are failing)

File: src/client/renderingengine.cpp
Comment (sfan5) [inline src/client/renderingengine.cpp:199]: this file is not installed on windows so this doesn't work currently

would make sense to move it to `textures/base/pack` imo

File: src/client/renderingengine.cpp
Comment (sfan5) [inline src/client/renderingengine.cpp:204]: `ICON_DIR` is also unset on windows

**Review by sfan5 — CHANGES_REQUESTED** [review]

File: src/client/renderingengine.cpp
Comment (Desour) [reply src/client/renderingengine.cpp]: We already have a `logo.png` file there. It's 100x100 pixels, which is maybe not a good size. But using it seems to work for me.

**Comment by sfan5** [comment]
No longer needs to wait.

**Review by nerzhul — APPROVED** [review]

**Comment by nerzhul** [comment]
clearly a good move

**Comment by sfan5** [comment]
Oops, yes.

**Comment by Desour** [comment]
Does somebody want to do more testing (i.e. on win, mac, android, or non-android touch)?
If not, I'll merge soon-ish.

**Pull request merged**
